vault/auth.go

```diff
@@ -457,6 +461,9 @@ func (c *Core) teardownCredentials() error {
 // newCredentialBackend is used to create and configure a new credential backend by name
 func (c *Core) newCredentialBackend(
 	t string, sysView logical.SystemView, view logical.Storage, conf map[string]string) (logical.Backend, error) {
+	if alias, ok := credentialAliases[t]; ok {
```

Comment:
Vishal and I talked about this yesterday after he commented, I think that the right thing to do is not to change the type, so that anyone checking mount tables for expected values don't suddenly have the rug pulled out from under them.

My thinking is we should allow "aws-ec2" to be an auth type that can be enabled, and for any such mount types, keep the type value as such. However, all docs and such going forward should use "aws" and any mount created as that type should read "aws" for the type.